utils/defined_error.h

```diff
@@ -23,7 +23,8 @@ typedef enum {
   RET_HTTP_CERT,
   RET_HTTP_CONNECT,
   RET_HTTP_SSL,
-} http_retcode_t;
+  RET_HARDWARE_DEPS,
```

Comment:
What is `RET_HARDWARE_DEPS`?

Reply:
My failed.

Reply:
Always write down meaningful comment and enumerations in header file.